core/src/main/java/org/apache/calcite/avatica/util/AbstractCursor.java

```diff
@@ -955,10 +955,12 @@ protected Number getNumber() throws SQLException {
    */
   static class TimeFromNumberAccessor extends NumberAccessor {
     private final Calendar localCalendar;
+    private final int precision;
 
-    TimeFromNumberAccessor(Getter getter, Calendar localCalendar) {
+    TimeFromNumberAccessor(Getter getter, Calendar localCalendar, int precision) {
       super(getter, 0);
```

Comment:
I have a question, should the 2 in `Time(2)` be `scale` or `precision`?

Reply:
I will rename it to "scale"

Reply:
I guess I have to leave it as `precision`, since both the columnMetadataFields which originate this value call it `precision`, and the helper functions that manipulate this field also call it `precision` (as you point out in your other comment).  It would be too confusing to use different names for this value in different parts of the codebase. 

I propose we file a new issue if we want to modify the field name to `scale`, to make sure that we cover all the software layers involved. In the meantime it would be great to merge this PR because it fixes a genuine bug.